client/app/admin/admin.controller.js

```diff
@@ -2,46 +2,74 @@
 
 export default class AdminController {
   /*@ngInject*/
-  constructor($scope, $http, $route, User, Upload) {
-    $scope.users = User.query();
+  constructor($http, $route, User, Upload, Settings, ENV, CAS_ENABLED, LOCAL_LOGIN_ENABLED, DEBUG_LOGIN_ENABLED) {
+    this.users = User.query();
     this.User = User;
     this.$http = $http;
+    this.Upload = Upload;
     this.$route = $route;
-    $scope.uploadUserCSV = function(file){
+    this.sortorder = 'lastName';
+    this.settings = {};
 
-      if (!file){
-        console.log("null file");
-        return;
-      }
+    this.devMode = ENV == "development";
+    this.CAS_ENABLED = CAS_ENABLED;
+    this.LOCAL_LOGIN_ENABLED = LOCAL_LOGIN_ENABLED;
+    this.DEBUG_LOGIN_ENABLED = DEBUG_LOGIN_ENABLED;
+
+    Settings.current({},(response) => {
+      this.settings = response;
+    },(error) => {
+      this.settings = {};
+    });
+  }
+
+  toggleLogin(type){
+    var update = {};
+    update[type] = !this.settings.login[type]
+    this.$http.put('/api/settings/login', update)
+    .success(() => {
+      this.settings.login[type] = !this.settings.login[type]
+    }).error(() => {
+    });
+  }
 
-      Upload.upload({
-          url: '/api/users/csv',
-          data: {
-            files: [file]
-          },
-          objectKey: '.k',
-          arrayKey: '[i]'
-      }).success(((response) => {
-        $scope.csvResults = response.join("<br/>");
-        $scope.users = User.query();
-        $scope.users.sort(( a , b) => {
-          return a.lastName > b.lastName;
-        })
-      }));
+  uploadUserCSV(file){
+    if (!file){
+      console.log("null file");
```

Comment:
:penguin: i agree that there should be a log statement, but it might be hard to find this in the code in the future- i think a more verbose log is a good thing here- but very minor concern